zipline/utils/events.py

```diff
@@ -466,18 +466,15 @@ def should_trigger(self, dt, env):
             # of the simulation start
             self.calculate_start_and_end(dt, env)
 
-            # If we've missed the first trigger because it occurs before the
-            # simulation starts, recalculate for the next week
-            if dt > self.next_date_end:
-                self.calculate_start_and_end(dt + datetime.timedelta(days=7),
-                                             env)
-
-        # if the given dt is within the next matching day, return true. Also
-        # calculate the start and end dates for the next trigger
+        # If we've passed the trigger, calculate the next one
+        if dt > self.next_date_end:
```

Comment:
In the previous PR, we had considered changing `calculate_start_and_end` to make this guarantee for the consumer by adding this condition to the while loop.  Do you think that would be any cleaner now?